src/components/NewPost/NewPost.tsx

```diff
@@ -0,0 +1,144 @@
+import React, { useState, useEffect } from "react";
+import { CustomFormInput } from "../CustomInputForm/CustomInputForm";
+import styled from "styled-components";
+import { CustomSubmitButton } from "../CustomButtons/CustomSubmitButton/CustomSubmitButton";
+import { CustomBackButton } from "../CustomButtons/CustomBackButton/CustomBackButton";
+import { apiEndpoints, PUBLIC_V1 } from "../../configs/apiEndpoints";
+import { appRoutesList } from "../../configs/appRoutesList";
+import { AlertMessage } from "../AlertMessage/AlertMessage";
+import { NoDataFound } from "../NoDataFound/NoDataFound";
+import { customFetch } from "../../customFunctions/customFetch";
+
+const StyledH1 = styled.h1`
+  text-align: center;
+`;
+
+const StyledSelect = styled.select`
+  width: 90%;
+  height: 35px;
+  background: ${(props) => props.theme.colors.white};
+  color: ${(props) => props.theme.colors.gray};
+  font-size: 14px;
+  border: none;
+  font-size: 16px;
+  font-family: "Poppins";
+`;
+
+type UsersProps = {
+  id: number;
+  name: string;
+};
+
+type Users = {
+  users: UsersProps[];
+};
+
+export const NewPost = () => {
+  const [postData, setPostData] = useState({
+    userId: "",
+    title: "",
+    body: "",
+  });
+  const [allUsers, setAllUsers] = useState<Users>({ users: [] });
+  const [statusMessage, setStatusMessage] = useState<number>(0);
+  const [displayedToastr, setDisplayedToastr] = useState(false);
+  const [responseFetchError, setResponseFetchError] = useState(false);
+
+  useEffect(() => {
+    setTimeout(() => {
+      setDisplayedToastr(false);
+    }, 3000);
+    const handleGetResponse = async () => {
+      try {
+        const response = await customFetch(apiEndpoints.USERS, "GET");
+        if (response.ok === true) {
+          const resJson = await response.json();
+          setAllUsers({ users: resJson.data });
+        } else {
+          setAllUsers({ users: [] });
+          setResponseFetchError(true);
+        }
+      } catch (error) {
+        setResponseFetchError(true);
+      }
+    };
+    handleGetResponse();
+  }, [displayedToastr]);
+
+  const handleSubmit = async (event: React.FormEvent<EventTarget>) => {
+    const handlePostRequest = async () => {
+      const response = await customFetch(
+        `${PUBLIC_V1}/users/${postData.userId}/posts`,
```

Comment:
vyextrahovat url

Reply:
pridaná fcia pre vrátenie url, každopádne v nasledujúcich PR pri pracovaní s ďalšími url to budem musieť zjednotiť, takže to ešte neskôr budem prepracovávať a počítam s tým